### PR TITLE
Check typo & data-table matches tests

### PR DESCRIPTION
**Pull request by marcysutton**
There were two issues addressed here, but one was already fixed (#561). The other is a single word change in a check description.

Closes https://github.com/dequelabs/axe-core/issues/526

### CODE REVIEW

File: lib/checks/tables/th-has-data-cells.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ var tableUtils = axe.commons.table;
 var cells = tableUtils.getAllCells(node);
 var checkResult = this;
 
+var exemptRoles = ['presentation', 'none'];
+if (node.hasAttribute('role') && exemptRoles.includes(node.getAttribute('role'))) {
+	return true;
```
Comment (WilcoFiers) [inline lib/checks/tables/th-has-data-cells.js]: These tables should be ignored, not passed. I suggest putting this in the selector or matcher instead.

File: doc/rule-descriptions.md [hunk 2]
```diff
@@ -15,7 +15,7 @@
 | blink | Ensures &lt;blink&gt; elements are not used | cat.time-and-media, wcag2a, wcag222, section508, section508.22.j | true |
 | button-name | Ensures buttons have discernible text | cat.name-role-value, wcag2a, wcag412, section508, section508.22.a | true |
 | bypass | Ensures each page has at least one mechanism for a user to bypass navigation and jump straight to the content | cat.keyboard, wcag2a, wcag241, section508, section508.22.o | true |
-| checkboxgroup | Ensures related &lt;input type=&quot;checkbox&quot;&gt; elements have a group and that that group designation is consistent | cat.forms, best-practice | true |
+| checkboxgroup | Ensures related &lt;input type=&quot;checkbox&quot;&gt; elements have a group and that group designation is consistent | cat.forms, best-practice | true |
```
Comment (dylanb) [inline doc/rule-descriptions.md]: Actually, this is not a typo, perhaps you could change it to

```
Ensures related &lt;input type=&quot;checkbox&quot;&gt; elements have a group and that the group designation is consistent
```

Comment (dylanb) [reply doc/rule-descriptions.md]: Clarification: there are two sentences here, individually, they could read

Ensures related input elements have a group and
Ensures that the group designation is consistent

File: lib/checks/tables/th-has-data-cells.js (hunk 1 above)
Comment (dylanb) [reply lib/checks/tables/th-has-data-cells.js]: agree

Comment (marcysutton) [reply lib/checks/tables/th-has-data-cells.js]: Ok, I've updated it. Thanks!

**Comment by marcysutton** [comment]
This PR has been updated with your feedback and is ready for another review.

File: test/rule-matches/data-table-matches.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+describe('data-table-matches', function () {
```
Comment (WilcoFiers) [inline test/rule-matches/data-table-matches.js]: This test should have a "should return false" case as well.

Comment (marcysutton) [reply test/rule-matches/data-table-matches.js]: Added, and I fixed the typos in the first two test names.

File: lib/rules/data-table-matches.js [hunk 4]
```diff
@@ -1 +1,5 @@
+var exemptRoles = ['presentation', 'none'];
+if (node.hasAttribute('role') && exemptRoles.includes(node.getAttribute('role'))) {
+	return false;
+}
 return axe.commons.table.isDataTable(node);
```
[the remark this answers is not in the document]
Comment (marcysutton) [reply lib/rules/data-table-matches.js]: That commons function also checks if the table is focusable, does that still apply?
```js
// is-data-table.js
// The element is not focusable and has role=presentation
if ((role === 'presentation' || role === 'none') && !dom.isFocusable(node)) {
    return false;
}
```

**Comment by marcysutton** [comment]
I kept the data-table-matches tests but the underlying issue in #561 was already fixed. This is now a very minor PR and is ready to be merged.

**Pull request merged**